### PR TITLE
Added Minimum Absolute Difference in Array problem [C and C++ implementation]

### LINKED ISSUE #1776: Minimum Absolute Difference in Array [C and C++]
I am interested in working on this issue. Kindly assign it to me.

### PR DESCRIPTION
**Pull request by apoorvam25**
Fixes #1776 

 Checklist:
-  [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

 Changes proposed in this pull request:
-  Added implementation of the problem in C and C++. 

 Languages Used: 
- C 
- C++

 Files Added:
- MinAbsolute_Diff.c
- MinAbsolute_Diff.cpp

### CODE REVIEW

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
+//Given an integer array A of size N, find and return the minimum absolute difference between any two elements in the array.
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: break this into two sentences and make it as multi line commend

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+//The absolute difference between two elements ai, and aj (where i != j ) is |ai - aj|.
+
+#include <stdio.h>
+#include<stdlib.h>
+#include<limits.h>
+
+int cmpfunc (const void * a, const void * b) {
+   return ( *(int*)a - *(int*)b );
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: please correct the indentation (keep four spaces )

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+int minAbsoluteDiff(int arr[], int n) {
+
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: remove this line

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    qsort(arr, n, sizeof(int), cmpfunc);
+    int mindiff= INT_MAX;
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: keep space before and after =, correct the indentation

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    for(int i=0;i<n-1;i++){
+        if(abs(arr[i]-arr[i+1])<mindiff)
+            mindiff = abs(arr[i]-arr[i+1]);
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: keep space before and after arithmetic operators, correct the indentation

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: remove this line

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    return mindiff;
+
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: remove this line too

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+int cmpfunc (const void * a, const void * b) {
+   return ( *(int*)a - *(int*)b );
+}
+
+int minAbsoluteDiff(int arr[], int n) {
+
+    qsort(arr, n, sizeof(int), cmpfunc);
+    int mindiff= INT_MAX;
+    for(int i=0;i<n-1;i++){
+        if(abs(arr[i]-arr[i+1])<mindiff)
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: add a comment here

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+            mindiff = abs(arr[i]-arr[i+1]);
+    }
+
+    return mindiff;
+
+}
+
+int main() {
+
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: remove this line

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+	int size,i;
+	scanf("%d",&size);
+	int input[size];
+
+	for(i = 0; i < size; i++)
+		scanf("%d",&input[i]);
+
+	printf("%d",minAbsoluteDiff(input,size));
+
+	return 0;
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: correct the indentation

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+            mindiff = abs(arr[i]-arr[i+1]);
+    }
+
+    return mindiff;
+
+}
+
+int main() {
+
+	int size,i;
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: keep a space after comma

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+	scanf("%d",&size);
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: keep a space after comma

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+	int input[size];
+
+	for(i = 0; i < size; i++)
+		scanf("%d",&input[i]);
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: keep a space after comma

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.c [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+
+	printf("%d",minAbsoluteDiff(input,size));
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.c]: keep  space after comma

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp [hunk 15]
```diff
@@ -0,0 +1,44 @@
+//Given an integer array A of size N, find and return the minimum absolute difference between any two elements in the array.
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp]: break this into two lines and make it as multi line commend

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+//The absolute difference between two elements ai, and aj (where i != j ) is |ai - aj|.
+
+#include<bits/stdc++.h>
+#include <iostream>
+#include<algorithm>
+using namespace std;
+
+int minAbsoluteDiff(int arr[], int n)
+{
+
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp]: remove this line

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+    std::sort(arr,arr+n);
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp]: - keep space after comma

- keep space before and after +

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+    int mindiff= INT_MAX;
+    for(int i=0; i<n-1; i++)
+    {
+        if(abs(arr[i]-arr[i+1])<mindiff)
+            mindiff = abs(arr[i]-arr[i+1]);
+    }
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp]: keep space before and after operators

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp]: remove this line

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+    return mindiff;
+
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp]: remove this line too

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+int main()
+{
+
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp]: remove this line

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+    int size;
+    cin >> size;
+    int *input = new int[1 + size];
+
+    for(int i = 0; i < size; i++)
+        cin >> input[i];
+
+    cout<< minAbsoluteDiff(input,size) << endl;
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp]: - keep space before << 

- keep a space after comma

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp]: remove this line

File: MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+    return 0;
+
```
Comment (asha15) [inline MinAbsolute_Diff_Array/MinAbsolute_Diff.cpp]: remove this line too

**Comment by asha15** [comment]
change the folder name as Minimum_ Absolute_Difference_In_Array

File: Backtracking_Crossword_Problem/Backtracking_Problem.cpp [hunk 25]
```diff
@@ -0,0 +1,204 @@
+/*Given a 10*10 grid(character array) with '+' or '-' as its cell values.
```
Comment (jainaman224) [inline Backtracking_Crossword_Problem/Backtracking_Problem.cpp]: Remove this file as it is not related to issue.

File: Minimum_ Absolute_Difference_In_Array/MinAbsolute_Diff.c [hunk 26]
```diff
@@ -0,0 +1,44 @@
+/*Given an integer array A of size N, find and return
```
Comment (jainaman224) [inline Minimum_ Absolute_Difference_In_Array/MinAbsolute_Diff.c]: Fix folder and file name.

File: Minimum_ Absolute_Difference_In_Array/MinAbsolute_Diff.c [hunk 27]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+int main()
+{
+	int size,i;
+	scanf("%d",&size);
+	int input[size];
+
+	for(i = 0; i < size; i++)
+		scanf("%d", &input[i]);
```
Comment (jainaman224) [inline Minimum_ Absolute_Difference_In_Array/MinAbsolute_Diff.c]: Incorrect indentation.

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c [hunk 28]
```diff
@@ -0,0 +1,42 @@
...
+int cmpfunc (const void * a, const void * b)
+{
+    return ( *(int*)a - *(int*)b );
+}
+
+int minAbsoluteDiff(int arr[], int n)
+{
+    qsort(arr, n, sizeof(int), cmpfunc);
+    int mindiff = INT_MAX;
+    for(int i=0;i<n-1;i++)
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c]: Please add space around the operators.

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c [hunk 29]
```diff
@@ -0,0 +1,42 @@
...
+    {
+       if(abs(arr[i] - arr[i+1]) < mindiff)
+          mindiff = abs(arr[i] - arr[i+1]);
+    }
+    return mindiff;
+}
+
+int main()
+{
+	int size,i;
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c]: Please add space after a comma.

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c [hunk 30]
```diff
@@ -0,0 +1,42 @@
...
+	scanf("%d",&size);
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c]: Please add space after a comma.

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c [hunk 31]
```diff
@@ -0,0 +1,42 @@
...
+	int input[size];
+	for(i = 0; i < size; i++)
+       scanf("%d", &input[i]);
+
+	printf("%d",minAbsoluteDiff(input,size));
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c]: Please add space after a comma.

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.cpp [hunk 32]
```diff
@@ -0,0 +1,38 @@
...
+#include<bits/stdc++.h>
+#include <iostream>
+#include<algorithm>
+using namespace std;
+
+int minAbsoluteDiff(int arr[], int n)
+{
+    std::sort(arr, arr+n);
+    int mindiff = INT_MAX;
+    for(int i=0; i<n-1; i++)
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.cpp]: Please add space around the operators.

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c [hunk 33]
```diff
@@ -0,0 +1,42 @@
...
+    return mindiff;
+}
+
+int main()
+{
+	int size,i;
+	scanf("%d",&size);
+	int input[size];
+	for(i = 0; i < size; i++)
+       scanf("%d", &input[i]);
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c]: Please correct the indentation.

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.cpp [hunk 34]
```diff
@@ -0,0 +1,38 @@
...
+    {
+       if(abs(arr[i] - arr[i+1]) < mindiff)
+        mindiff = abs(arr[i] - arr[i+1]);
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.cpp]: Please correct the indentation.

**Comment by somya-kapoor** [comment]
@apoorvam25 Please squash the commits.

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.cpp [hunk 35]
```diff
@@ -0,0 +1,38 @@
...
+using namespace std;
+
+int minAbsoluteDiff(int arr[], int n)
+{
+    std::sort(arr, arr + n);
+    int mindiff = INT_MAX;
+    for(int i = 0; i < n-1; i++)
+    {
+        if(abs(arr[i] - arr[i+1] ) < mindiff)
+            mindiff = abs(arr[i] - arr[i+1]);
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.cpp]: Add space around +

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c [hunk 36]
```diff
@@ -0,0 +1,42 @@
...
+    return ( *(int*)a - *(int*)b );
+}
+
+int minAbsoluteDiff(int arr[], int n)
+{
+    qsort(arr, n, sizeof(int), cmpfunc);
+    int mindiff = INT_MAX;
+    for(int i = 0; i < n - 1; i++)
+    {
+        if(abs(arr[i] - arr[i+1]) < mindiff)
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c]: Add space around +

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c [hunk 37]
```diff
@@ -0,0 +1,42 @@
...
+            mindiff = abs(arr[i] - arr[i+1]);
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c]: Add space around +

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.cpp [hunk 38]
```diff
@@ -0,0 +1,38 @@
...
+#include<algorithm>
+using namespace std;
+
+int minAbsoluteDiff(int arr[], int n)
+{
+    std::sort(arr, arr + n);
+    int mindiff = INT_MAX;
+    for(int i = 0; i < n-1; i++)
+    {
+        if(abs(arr[i] - arr[i+1] ) < mindiff)
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.cpp]: Add space around +

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.cpp [hunk 39]
```diff
@@ -0,0 +1,38 @@
...
+#include<bits/stdc++.h>
+#include <iostream>
+#include<algorithm>
+using namespace std;
+
+int minAbsoluteDiff(int arr[], int n)
+{
+    std::sort(arr, arr + n);
+    int mindiff = INT_MAX;
+    for(int i = 0; i < n-1; i++)
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.cpp]: Add space around -

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c [hunk 40]
```diff
@@ -0,0 +1,42 @@
...
+the minimum absolute difference between any two elements in the array.
+The absolute difference between two elements ai, and aj (where i != j ) is |ai - aj|*/
+
+#include<stdio.h>
+#include<stdlib.h>
+#include<limits.h>
+
+int cmpfunc (const void * a, const void * b)
+{
+    return ( *(int*)a - *(int*)b );
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c]: remove a space after( and before )

**Comment by somya-kapoor** [comment]
@apoorvam25 After making the changes, please squash the commits. Then it will ready to merge.

**Comment by somya-kapoor** [comment]
@apoorvam25 Are you working on it?

**Comment by apoorvam25** [comment]
Yes, I'm working on it.
 I'm not able to understand how commits are to be squashed. So It is taking
a bit longer than expected.

File: Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c [hunk 41]
```diff
@@ -0,0 +1,42 @@
+/*Given an integer array A of size N, find and return
```
Comment (somya-kapoor) [inline Minimum_ Absolute_Difference_In_Array/Minimum_Absolute_Difference_In_Array.c]: Please remove extra space in the file name ( after Minimum_ A...)
